src/GhciFind.hs

```diff
@@ -49,12 +83,30 @@ findCompletions infos fp sample sl sc el ec =
              return (Left ("No module info for the current file! Try loading it?"))
            Just moduleInf ->
              do df <- getDynFlags
-                let toplevelNames =
-                      fromMaybe [] (modInfoTopLevelScope (modinfoInfo moduleInf))
+                (qual, ident, minfs) <-
+                   let noQual = ("", sample, [modinfoInfo moduleInf])
+                       getModInfo qmname =
+                         findModule qmname Nothing >>= getModuleInfo
+                   in if '.' `elem` sample
+                      then do let importDecls = importDeclsFromInfo moduleInf
+                              case findQualifiedSource
+                                      importDecls
+                                      sample of
+                                 Just (qual, ident, qualModNames) -> do
+                                   minfos <- fmap catMaybes
+                                                  (mapM getModInfo qualModNames)
+                                   if null minfos
+                                      then return noQual
+                                      else return (qual, ident, minfos)
+                                 Nothing -> return noQual
+                      else return noQual
+                let toplevelNames = concat (mapMaybe modInfoTopLevelScope minfs)
                     filteredToplevels =
-                      filter (isPrefixOf sample)
```

Comment:
I think with `ideclAs :: Maybe ModuleName` from `ImportDecl` we can just change this to a `mapMaybe matchPrefix` where `matchPrefix :: [ImportDecl] -> String -> Name -> Maybe String` which would yield e.g. `matchPrefix .. "M.ins" "insertWith"` -> `Just "M.insertWith"` and `matchPrefix .. "sor" "sortBy" -> Just "sortBy"`.